.github/workflows/ci-cd.yml

```diff
@@ -46,7 +56,7 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
           fetch-depth: 0
 
-      - if: steps.conditionals_handler.outputs.is_default_branch == 'true'
+      - if: steps.conditionals_handler.outputs.is_push_to_default_branch == 'true'
```

Comment:
Used the newly added `is_push_to_default_branch` condition, instead of the `is_default_branch`, to further restrict the new tag generating step to only execute when a push commit to the default branch is made.

Reply:
I'm somewhat confused by this process.  First, the default branch is vnext, not master.  We don't create releases when we are updating vnext.  Also, we don't push directly to vnext, we generally PR to it.  What am I missing?

Reply:
>  We don't create releases when we are updating vnext

If you are not making releases from your default branch (vnext), we will change it to the correct one. Is it `master` or another branch ?

Also not every push commit creates a new GitHub tag and release. In the [documentation](https://github.com/microsoft/OpenAPI.NET/blob/vnext/docs/CI-CD_DOCUMENTATION.md#3-automated-versioning) we submitted with the last PR, you can check out the exact prerequisite to trigger an output (meaning create a new tag) from the "Bump GH tag" step, which in turn triggers a GitHub release.

The path is this:

Push commit in release branch with SemVer label 
-> "Bump GH tag" step creates a new GitHub tag and pushes it to the repo 
-> CD job evaluates if there is a new GitHub tag and if there is, executes
-> the "Create and publish release" step in the CD job uses the newly created and pushed GitHub tag to create a GitHub release

> we generally PR to it

The merge of a PR triggers a push commit, so you can either create a release though your release branch, by committing into it or merging a PR.

If you have any other questions about the automated release process, please let us know! 🙂

Reply:
We create releases off the master branch.

Reply:
@darrelmiller with the last commit:

- switch the release branch from default_branch to master
- remove the data gatherer and conditionals handler steps, since they were used specifically for the default_branch as release branch logic

In the current state of the pipeline if there are no errors present, a push commit or PR merge to the master branch with a SemVer label in the title, would deploy (publish packages and create a GitHub release).

Push commits or PR merges to the default_branch would trigger the pipeline, but not deploy.